stake-pool/program/src/processor.rs

```diff
@@ -293,70 +294,100 @@ impl Processor {
         // Clock sysvar account
         let clock_info = next_account_info(account_info_iter)?;
         let clock = &Clock::from_account_info(clock_info)?;
+        // Rent sysvar account
+        let rent_info = next_account_info(account_info_iter)?;
+        let rent = &Rent::from_account_info(rent_info)?;
         // Token program ID
         let token_program_info = next_account_info(account_info_iter)?;
 
+        // Check if transaction was signed by owner
+        if !owner_info.is_signer {
+            return Err(StakePoolError::SignatureMissing.into());
+        }
+
+        let mut stake_pool = StakePool::deserialize(&stake_pool_info.data.borrow())?;
         // Stake pool account should not be already initialized
-        if State::Unallocated != State::deserialize(&stake_pool_info.data.borrow())? {
+        if stake_pool.is_initialized() {
             return Err(StakePoolError::AlreadyInUse.into());
         }
 
         // Check if validator stake list storage is unitialized
         let mut validator_stake_list =
             ValidatorStakeList::deserialize(&validator_stake_list_info.data.borrow())?;
-        if validator_stake_list.is_initialized {
+        if validator_stake_list.is_initialized() {
             return Err(StakePoolError::AlreadyInUse.into());
         }
-        validator_stake_list.is_initialized = true;
+        validator_stake_list.version = ValidatorStakeList::VALIDATOR_STAKE_LIST_VERSION;
```

Comment:
nit: What's the reason of having both VALIDATOR_STAKE_LIST_VERSION and PROGRAM_VERSION?  I might be missing a use case, or perhaps you want to keep things as flexible as possible.  I'm worried about handling version mismatches later on

Reply:
I had in mind that both these structures might change format independently and in this case only one of them has to be migrated. For example if `ValidatorStakeList` changes there will be no need to migrate pool state.